### PR TITLE
Replace actions/setup-ruby with ruby/setup-ruby

### PR DESCRIPTION
**Pull request by sloshy**
As per: https://github.com/actions/setup-ruby/issues/80

Replaces actions/setup-ruby with ruby/setup-ruby as well as a way to define the default ruby version via the `.ruby-version` file. I looked into this due to noticing that there was a ruby-related build issue in sbt-microsites that I was not having on my local machine, and traced it down to a ruby syntax error. This will make sure that builds using ruby are all using the same latest-ish (v2 series, not v3) version that I am and eliminate that possible inconsistency.

### CODE REVIEW

File: root-files/.ruby-version [hunk 1]
```diff
@@ -0,0 +1 @@
+2.7.2
```
Comment (alejandrohdezma) [inline root-files/.ruby-version:1]: Shouldn't this file be added in each individual repository using Ruby?

**Review by alejandrohdezma — DISMISSED** [review]

File: root-files/.ruby-version (hunk 1 above)
Comment (sloshy) [reply root-files/.ruby-version:1]: If ruby is part of the standard CI workflow defined here, then, wouldn't that be every repository?

Comment (sloshy) [reply root-files/.ruby-version:1]: I'm going to merge this for now, seeing as you approved it and I might need this to get a passing build, but, please feel free to respond or continue this discussion here or elsewhere if you see there's a better way to set this up.

Comment (sloshy) [reply root-files/.ruby-version:1]: Ah, this reminded me: I forgot to symlink this to the root of this repository, which I did just now. Sorry about that.

**Review by juanpedromoreno — APPROVED** [review]

File: root-files/.ruby-version (hunk 1 above)
Comment (alejandrohdezma) [reply root-files/.ruby-version:1]: Yeah, that is right 😓 

Thank you very much @sloshy!

**Pull request merged**